### PR TITLE
#1730 Update unit test with new api and snackbar

### PR DESCRIPTION
**Pull request by wu87988622**
### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #1730 )
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [x] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

**Comment by konekoya** [comment]
retry manager-e2e

File: ohara-manager/client/src/components/pages/WorkspacesPage/WorkspacesDetailPage/__tests__/StreamApp.test.js [hunk 1]
```diff
@@ -15,46 +15,56 @@
...
+
+    jest.spyOn(useApi, 'useUploadApi').mockImplementation(() => {
+      return {
+        getData: jest.fn(),
+      };
+    });
+
+    jest.spyOn(useApi, 'useFetchApi').mockImplementation(() => {
+      return {
+        data: {
```
Comment (konekoya) [inline ohara-manager/client/src/components/pages/WorkspacesPage/WorkspacesDetailPage/__tests__/StreamApp.test.js]: Hmm, is it possible that we can rename one of the data here?

File: ohara-manager/client/src/components/pages/WorkspacesPage/__tests__/WorkspacesPage.test.js [hunk 2]
```diff
@@ -86,13 +100,16 @@ describe.skip('<WorkspacesPage />', () => {
...
-        result: [...workers],
-      },
-    };
-
-    fetchWorkers.mockImplementation(() => Promise.resolve(res));
+    jest.spyOn(useApi, 'useFetchApi').mockImplementation(() => {
+      return {
+        data: {
+          data: {
+            result: [...workers],
```
Comment (konekoya) [inline ohara-manager/client/src/components/pages/WorkspacesPage/__tests__/WorkspacesPage.test.js]:
```suggestion
            result: workers,
```

File: ohara-manager/client/src/components/pages/NodePage/__tests__/NodeListPage.test.js [hunk 3]
```diff
@@ -14,92 +14,116 @@
...
+                name: generate.serviceName(),
+                services: [
+                  {
+                    name: generate.serviceName(),
+                    clusterNames: [{ clusterName: generate.serviceName() }],
+                  },
+                ],
+                user: generate.userName(),
+                port: generate.port(),
+                password: generate.serviceName(),
```
Comment (konekoya) [inline ohara-manager/client/src/components/pages/NodePage/__tests__/NodeListPage.test.js]:
```suggestion
                password: generate.password(),
```

File: ohara-manager/client/src/components/pages/NodePage/__tests__/NodeListPage.test.js [hunk 4]
```diff
@@ -227,19 +258,15 @@ describe.skip('<NodeListPage />', () => {
       expect(saveButton).toBeDisabled();
     });
 
-    it('enables save button when there is new change made in the form', async () => {
-      const {
-        getByText,
-        getByTestId,
-        getByPlaceholderText,
-      } = await waitForElement(() => render(<NodeListPage />));
+    it.skip('enables save button when there is new change made in the form', async () => {
```
Comment (konekoya) [inline ohara-manager/client/src/components/pages/NodePage/__tests__/NodeListPage.test.js]: Can we add some comment here and leave an issue number so we can come back and fix this

File: ohara-manager/client/src/components/pages/WorkspacesPage/WorkspacesDetailPage/__tests__/Nodes.test.js [hunk 5]
```diff
@@ -15,64 +15,71 @@
...
-          nodeNames,
-          brokerClusterName: generate.serviceName(),
-        },
-      },
-    };
-
-    fetchWorker.mockImplementation(() => Promise.resolve(workerRes));
-    fetchNodes.mockImplementation(() => Promise.resolve(nodesRes));
+    jest.spyOn(useApi, 'useGetApi').mockImplementation(() => {
+      return { getData: () => {} };
```
Comment (konekoya) [inline ohara-manager/client/src/components/pages/WorkspacesPage/WorkspacesDetailPage/__tests__/Nodes.test.js]: Why not `jest.fn`?

File: ohara-manager/client/src/components/pages/WorkspacesPage/WorkspacesDetailPage/__tests__/Nodes.test.js [hunk 6]
```diff
@@ -95,30 +102,32 @@ describe.skip('<Nodes />', () => {
...
-
-    fetchWorker.mockImplementation(() => Promise.resolve(workerRes));
-    fetchNodes.mockImplementation(() => Promise.resolve(nodesRes));
-
-    const res = { data: { isSuccess: true, result: nodes } };
-    fetchNodes.mockImplementation(() => Promise.resolve(res));
-
-    const { getAllByTestId } = await render(<Nodes {...props} />);
+    jest.spyOn(useApi, 'useFetchApi').mockImplementation(url => {
+      if (url === URL.WORKER_URL + '/' + props.workspaceName) {
```
Comment (konekoya) [inline ohara-manager/client/src/components/pages/WorkspacesPage/WorkspacesDetailPage/__tests__/Nodes.test.js]:
```suggestion
      if (url === `${URL.WORKER_URL}/${props.workspaceName}`) {
```

File: ohara-manager/client/src/components/pages/NodePage/__tests__/NodeListPage.test.js (hunk 4 above)
Comment (wu87988622) [reply ohara-manager/client/src/components/pages/NodePage/__tests__/NodeListPage.test.js]: ok

File: ohara-manager/client/src/components/pages/WorkspacesPage/WorkspacesDetailPage/__tests__/Nodes.test.js (hunk 5 above)
Comment (wu87988622) [reply ohara-manager/client/src/components/pages/WorkspacesPage/WorkspacesDetailPage/__tests__/Nodes.test.js]: I will fix this sorry

File: ohara-manager/client/src/components/pages/WorkspacesPage/WorkspacesDetailPage/__tests__/StreamApp.test.js (hunk 1 above)
Comment (wu87988622) [reply ohara-manager/client/src/components/pages/WorkspacesPage/WorkspacesDetailPage/__tests__/StreamApp.test.js]: first data is fn return key
second data is response body key

**Pull request merged**